workers/pipeline/pipeline.go

```diff
@@ -174,62 +169,43 @@ func (ap *ActivePipelines) Replace(p gaia.Pipeline) bool {
 
 // ReplaceByName replaces the pipeline that has the given name with the given pipeline.
 func (ap *ActivePipelines) ReplaceByName(n string, p gaia.Pipeline) bool {
-
-	var index int
-	var pipelineIndex int
-	var found bool
-
-	for pipeline := range ap.Iter() {
+	for index, pipeline := range ap.Iter() {
 		if pipeline.Name == n {
-			found = true
-			pipelineIndex = index
+			ap.Update(index, p)
+			return true
 		}
-		index++
-	}
-
-	if found {
-		ap.Update(pipelineIndex, p)
 	}
-
-	return found
-
+	return false
 }
 
 // Iter iterates over the pipelines in the concurrent slice.
-func (ap *ActivePipelines) Iter() <-chan gaia.Pipeline {
-	c := make(chan gaia.Pipeline)
-
-	go func() {
-		ap.RLock()
-		defer ap.RUnlock()
-		for _, pipeline := range ap.Pipelines {
-			c <- pipeline
-		}
-		close(c)
-	}()
-
+func (ap *ActivePipelines) Iter() []gaia.Pipeline {
```

Comment:
`Iter` sounds like you get an iterator back. Maybe `GetAll` or `ToSlice` is better?

Reply:
@michelvocks I didn't wanted to change the whole thing at once... I thought I'll keep the name for now, to see if anything breaks.

But something I didn't realise is that I'm already giving back a list. Why am I going over it again in some situations. :D Made no sense. thanks for spotting that.

I'll rename it. :)